### PR TITLE
feat: Added sql database vulnerability scans

### PR DESCRIPTION
**Pull request by amanenk**
- added vulnerability scans
- sql mockgen splited into parts to make it more readable

### CODE REVIEW

File: resources/provider/migrations/timescale/5_v0.5.0.up.sql [hunk 1]
```diff
@@ -1,76 +1,116 @@
...
+    "storage_endpoint"           text,
+    "storage_account_access_key" text,
+    "retention_days"             integer,
+    "creation_time"              timestamp without time zone,
+    CONSTRAINT azure_sql_server_security_alert_policy_pk PRIMARY KEY (cq_fetch_date, server_cq_id, id),
+    UNIQUE (cq_fetch_date, cq_id)
 );
 CREATE INDEX ON azure_sql_server_security_alert_policy (cq_fetch_date, server_cq_id);
 SELECT setup_tsdb_child('azure_sql_server_security_alert_policy', 'server_cq_id', 'azure_sql_servers', 'cq_id');
+CREATE TABLE IF NOT EXISTS "azure_sql_database_db_vulnerability_assessment_scans"
```
Comment (roneli) [inline resources/provider/migrations/timescale/5_v0.5.0.up.sql:80]: these migrations should be in next version v0.6.0

File: resources/services/sql/databases.go [hunk 2]
```diff
@@ -10,281 +10,270 @@ import (
...
 				Name:        "sku_name",
-				Description: "The name of the SKU, typically, a letter + Number code, eg P3",
+				Description: "The name of the SKU, typically, a letter + Number code, e.g",
 				Type:        schema.TypeString,
 				Resolver:    schema.PathResolver("Sku.Name"),
 			},
 			{
 				Name:        "sku_tier",
-				Description: "The tier or edition of the particular SKU, eg Basic, Premium",
+				Description: "The tier or edition of the particular SKU, e.g",
```
Comment (roneli) [inline resources/services/sql/databases.go:40]: nit: revert descriptions

**Review by roneli — CHANGES_REQUESTED** [review]
A little hard to review with lots of shifting, looks good in general, noticed some descriptions going missing + ignore in tests missing (we should add ability to mark them when generating)

**Review by roneli — APPROVED** [review]
LGTM

File: resources/services/sql/databases.go [hunk 3]
```diff
@@ -429,10 +408,9 @@ func SQLDatabases() *schema.Table {
 				},
 			},
 			{
-				Name:        "azure_sql_database_db_threat_detection_policies",
-				Description: "Contains information about a database Threat Detection policy.",
-				Resolver:    fetchSqlDatabaseDbThreatDetectionPolicies,
-				Options:     schema.TableCreationOptions{PrimaryKeys: []string{"database_cq_id", "id"}},
```
Comment (disq) [inline resources/services/sql/databases.go:435]: Where did the PK go?

File: resources/services/sql/databases.go [hunk 4]
```diff
@@ -518,10 +548,9 @@ func SQLDatabases() *schema.Table {
 				},
 			},
 			{
-				Name:        "azure_sql_database_db_vulnerability_assessments",
-				Description: "DatabaseVulnerabilityAssessment a database vulnerability assessment",
-				Resolver:    fetchSqlDatabaseDbVulnerabilityAssessments,
-				Options:     schema.TableCreationOptions{PrimaryKeys: []string{"database_cq_id", "id"}},
```
Comment (disq) [inline resources/services/sql/databases.go:524]: Missing PK

File: resources/services/sql/databases.go [hunk 5]
```diff
@@ -328,50 +317,41 @@ func SQLDatabases() *schema.Table {
...
-				Description: "TransparentDataEncryption represents a database transparent data encryption configuration",
-				Type:        schema.TypeJSON,
-				Resolver:    resolveTransparentDataEncryption,
-			},
 		},
 		Relations: []*schema.Table{
 			{
 				Name:        "azure_sql_database_db_blob_auditing_policies",
-				Description: "Database blob auditing policy",
+				Description: "DatabaseBlobAuditingPolicy a database blob auditing policy",
```
Comment (disq) [inline resources/services/sql/databases.go:324]: Bad description?

**Pull request merged**